src/NHSD.BuyingCatalogue.Identity.Api/Controllers/AccountController.cs

```diff
@@ -92,20 +89,37 @@ public async Task<IActionResult> Login(LoginViewModel viewModel)
             if (Url.IsLocalUrl(returnUrl))
                 return Redirect(returnUrl);
 
-            if (string.IsNullOrEmpty(returnUrl))
+            if (string.IsNullOrWhiteSpace(returnUrl))
                 return Redirect("~/");
 
-            // User might have clicked on a malicious link - should be logged
-            throw new Exception("Invalid return URL");
+            // Return URL could be malicious
+            throw new InvalidReturnUrlException();
```

Comment:
Still not keen on throwing the exception here. It seems like we are using it just to provide a bit of logging. If that is the case I would rather use the logger than throw an exception. Would it not be better to redirect them to the error page instead?

We could get a second opinion on this. Obviously you have done the research. Most of the small samples I've seen do not raise this exception here.

Reply:
I think this a valid use of an exception – it's an exceptional case that we can't properly handle – and I think modelling that is its real purpose, with the added benefit that we log exceptions. 

We could just redirect to the error view but I don't think that would be as clear as throwing an exception. It would also require introducing more code into an already busy method. However, happy to get another opinion or two.